driver-mapping/src/main/java/com/datastax/driver/mapping/MappingManager.java

```diff
@@ -218,57 +219,33 @@ public Session getSession() {
 
     @SuppressWarnings("unchecked")
     private <T> Mapper<T> getMapper(Class<T> klass) {
-        Mapper<T> mapper = (Mapper<T>) mappers.get(klass);
-        if (mapper == null) {
-            synchronized (mappers) {
-                mapper = (Mapper<T>) mappers.get(klass);
-                if (mapper == null) {
-                    EntityMapper<T> entityMapper = AnnotationParser.parseEntity(klass, ReflectionMapper.factory(), this);
-                    mapper = new Mapper<T>(this, klass, entityMapper);
-                    Map<Class<?>, Mapper<?>> newMappers = new HashMap<Class<?>, Mapper<?>>(mappers);
-                    newMappers.put(klass, mapper);
-                    mappers = newMappers;
-                }
-            }
+        if (!mappers.containsKey(klass)) {
+            EntityMapper<T> entityMapper = AnnotationParser.parseEntity(klass, ReflectionMapper.factory(), this);
+            Mapper<T> mapper = new Mapper<T>(this, klass, entityMapper);
+            mappers.putIfAbsent(klass, mapper);
         }
-        return mapper;
+        return (Mapper<T>) mappers.get(klass);
```

Comment:
@avivcarmis `onTableRemoved` and `onTableChanged` should work on a best-effort basis. Honestly, if people out there are crazy enough to alter their schema while clients are issuing requests, it's their problem. I would actually suggest that we get rid of the `synchronized` blocks in the places you spotted and clearly specify in the documentation that concurrent modifications of the schema could end up in unexpected results.